public/js/Lineplot.js

```diff
@@ -0,0 +1,163 @@
+var d3 = require('d3');
+var _ = require('lodash');
+
+function Lineplot(sampleData, histogram) {
+    this.sampleData = sampleData;
+    this.featureLength = sampleData[0].feature_vector_std.length;
+    this.activeSample = 0;
+    this.activeFeature = 0;
+    this.histogram = histogram;
+
+    this.fullWidth = 600;
+    this.fullHeight = 400;
+
+    this.margin = {top: 20, right: 30, bottom: 20, left: 45};
+
+    this.width = this.fullWidth - this.margin.left - this.margin.right;
+    this.height = this.fullHeight - this.margin.top - this.margin.bottom;
```

Comment:
Ack, missed some of these as you pointed out earlier.. I'll replace the other 'magic' height and widths you've pointed out so they're like this one. I think height, widths and target elements could be defined as arguments to the plot constructor, so something like this

```
function Plot(data, width, height, element) {
    this.fullWidth = width
    // etc
}
```